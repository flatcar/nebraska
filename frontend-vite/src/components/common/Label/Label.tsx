import makeStyles from '@mui/styles/makeStyles';
import React from 'react';

const useStyles = makeStyles({
  label: {
    background: '#b4b4b4',
    color: '#ffffff',
    fontSize: '75%',
    textAlign: 'center',
    borderRadius: '.2em',
    padding: '.2em .6em .3em',
  },
});

export default function Label(props: { children: React.ReactNode }) {
  const classes = useStyles();
  return <span className={classes.label}>{props.children}</span>;
}
