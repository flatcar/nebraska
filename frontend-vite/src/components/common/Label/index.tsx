export { default } from './Label';
export * from './Label';
