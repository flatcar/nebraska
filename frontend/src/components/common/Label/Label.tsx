import { styled } from '@mui/material/styles';
import React from 'react';

const PREFIX = 'Label';

const classes = {
  label: `${PREFIX}-label`,
};

const Root = styled('span')({
  [`&.${classes.label}`]: {
    background: '#b4b4b4',
    color: '#ffffff',
    fontSize: '75%',
    textAlign: 'center',
    borderRadius: '.2em',
    padding: '.2em .6em .3em',
  },
});

export default function Label(props: { children: React.ReactNode }) {
  return <Root className={classes.label}>{props.children}</Root>;
}
