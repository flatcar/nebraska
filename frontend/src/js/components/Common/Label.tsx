import { makeStyles } from '@material-ui/core/styles';
import React from 'react';

const useStyles = makeStyles(theme => ({
  label: {
    background: '#b4b4b4',
    color: '#ffffff',
    fontSize: '75%',
    textAlign: 'center',
    borderRadius: '.2em',
    padding: '.2em .6em .3em',
  },
}));

export default function Label(props: {children: React.ReactChildren}) {
  const classes = useStyles();
  return (
    <span className={classes.label}>
      {props.children}
    </span>
  );
}
