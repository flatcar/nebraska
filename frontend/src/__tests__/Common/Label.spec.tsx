import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import Label from '../../components/common/Label';

describe('Label', () => {
  const minProps = {
    children: 'dummy label',
  };
  it('should render correct label', () => {
    const { asFragment, getByText } = render(<Label children={minProps.children} />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(minProps.children)).toBeTruthy();
  });
});
