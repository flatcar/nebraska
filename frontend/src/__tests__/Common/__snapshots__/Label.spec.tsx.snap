// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label > should render correct label 1`] = `
<DocumentFragment>
  <span
    class="Label-label css-1wzl15x"
  >
    dummy label
  </span>
</DocumentFragment>
`;
